groclient/lib.py

```diff
@@ -912,6 +913,122 @@ def reverse_geocode_points(access_token: str, api_host: str, points: list):
+
+    for entity in required_entities:
+        if entity not in series_object or not series_object[entity]:
+            raise ValueError(f"{entity} is required and supposed to be a positive integer.")
+
+    invalid_atts = set(series_object.keys()).difference(required_entities)
+    if len(invalid_atts):
+        raise ValueError(f"Unsupported fields: {invalid_atts}.")
+
+
+def generate_payload_for_v2_area_weighting(
+    series: Dict[str, int],
+    region_id: int,
+    weights: Optional[List[Dict[str, int]]] = None,
+    weight_names: Optional[List[str]] = None,
+    start_date: Optional[str] = None,
+    end_date: Optional[str] = None,
+    method: Optional[str] = "sum",
+):
+    payload = {
+        "region_id": region_id,
+        "method": method,
+    }
+
+    # validate series and weights selection
+    try:
+        validate_series_object(series)
+    except ValueError as error:
+        raise ValueError(f"Failed to parse series selection: {error}")
+    payload["series"] = series
+
+    if weights and len(weights):
+        if weight_names:
+            raise ValueError(f"weights and weight_names are mutually exclusive. Please specify only one.")
+        try:
+            for weight in weights:
+                validate_series_object(weight)
+        except ValueError as error:
+            raise ValueError(f"Failed to parse weight selections: {error}")
+        payload["weights"] = weights
+    else:
+        if not weight_names or not len(weight_names):
+            raise ValueError(f"Please specify either weights or weight_names in params.")
+        payload["weight_names"] = weight_names
+
+    # add optional attrs
+    if start_date:
+        payload["start_date"] = start_date
+    if end_date:
+        payload["end_date"] = end_date
+
+    return json.dumps(payload)
+
+
+def format_v2_area_weighting_response(response_content: Dict[str, Any]) -> pd.DataFrame:
+    try:
+        data_points = response_content["data_points"]
+        weighted_series_df = pd.DataFrame(data_points)
+
+        # convert unix timestamps and rename date cols
+        datetime_col_mappings = {
+            "start_date": "timestamp", # add start_date col which is equivalent to end_date
+            "end_date": "timestamp",
+            "available_date": "available_timestamp",
+        }
+        for new_col, col in datetime_col_mappings.items():
+            weighted_series_df[new_col] = pd.to_datetime(weighted_series_df[col], unit='s').dt.strftime('%Y-%m-%d')
+        weighted_series_df = weighted_series_df.drop(columns=datetime_col_mappings.values())
+
+        # append selected fields of series metadata
+        for key in ['item_id', 'metric_id', 'frequency_id', 'unit_id', 'source_id']:
+            weighted_series_df[key] = response_content["series_description"][key]
+
+        # append weights metadata as a single json
+        weighted_series_df["weights_metadata"] = json.dumps(response_content["weights_description"])
```

Comment:
not requested in the [snippet](https://colab.research.google.com/drive/1XvJ-gMtnG8yeB3V6EnwzRuspt2ptdk2g#scrollTo=ZXqN95QzWA35) 
let me know if we decide to remove the column
@muzigao-gro @sahuguet

Reply:
That's a good idea. If the customer does not need, they can simply ignore it.